### DISCUSSION TITLE
Syntax Error in TopicDataType.hpp When Generating SWIG Bindings for Fast DDS

### DISCUSSION THREAD

**Discussion by Jackie2chen** [eProsima/Fast-DDS-python#103] [Q&A]
Hello,

I am working with Fast DDS and attempting to generate Python bindings using SWIG version 4.2.1. However, I've run into a syntax error that seems to be related to the `TopicDataType.hpp` file.

Here's the error message I'm encountering:

TopicDataType.hpp:71: Error: Syntax error - possibly a missing semicolon (';')

fastcdr veriosn：2.0.0
fastrtps 2.12.0
swig 4.2.1

`find_package(SWIG REQUIRED)
include(${SWIG_USE_FILE})
if(${SWIG_VERSION} VERSION_GREATER_EQUAL 4)
  list(APPEND CMAKE_SWIG_FLAGS "-doxygen")
endif()

if(UNIX AND NOT APPLE)
  if (CMAKE_SIZEOF_VOID_P EQUAL 8)
    list(APPEND CMAKE_SWIG_FLAGS "-DSWIGWORDSIZE64")
  else()
    list(APPEND CMAKE_SWIG_FLAGS "-DSWIGWORDSIZE32")
  endif()
endif()
find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
include_directories(${Python3_INCLUDE_DIRS})
list(APPEND CMAKE_SWIG_FLAGS  "-DPY3")

# 指定IDL文件的位置
set(IDL_DIR "../idl")

# 指定生成的Python模块名称
set(MODULE_NAME "idl_python")

set(IDL_FILES
    "${IDL_DIR}/unit_test.i"
    "${IDL_DIR}/sim_time.i"
    "${IDL_DIR}/basestructure.i"
    "${IDL_DIR}/unit_testPubSubTypes.i"
    "${IDL_DIR}/sim_timePubSubTypes.i"
    "${IDL_DIR}/basestructurePubSubTypes.i"
)

set(CMAKE_CXX_STANDARD 17)
foreach(IDL_FILE ${IDL_FILES})
    get_filename_component(MODULE_FILE_NAME ${IDL_FILE} NAME_WE)
    set_source_files_properties(${IDL_FILE} PROPERTIES CPLUSPLUS ON
    USE_TARGET_INCLUDE_DIRECTORIES TRUE)
    swig_add_library(${MODULE_FILE_NAME}c
        TYPE SHARED
        LANGUAGE python
        OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/" # 指定输出目录
        SOURCES ${IDL_FILE})
    swig_link_libraries(${MODULE_FILE_NAME}c ${Python3_LIBRARIES} fastrtps ${TARGET_NAME})

    message(STATUS "Creating SWIG target: ${MODULE_FILE_NAME}c")
    set_property(TARGET ${MODULE_FILE_NAME}c_swig_compilation PROPERTY SWIG_COMPILE_DEFINITIONS SWIGWORDSIZE64 CXX_STANDARD 11)
    set_target_properties(${MODULE_FILE_NAME}c_swig_compilation PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/")
endforeach()`

**Comment by elianalf** [comment]
Hello, can you please provide all the necessary files to reproduce the error?

**Reply by Jackie2chen** [reply]
#ifndef KRIDER_IDL_INTERFACE_BASESTRUCTURE_BASESTRUCTURE
#define KRIDER_IDL_INTERFACE_BASESTRUCTURE_BASESTRUCTURE

module krider {
  module basestructure{
  struct BaseHeader {
      // uint64 timestamp_us;            //时间戳，以微秒为单位
      uint64 timestamp_ms;            //消息发送时间戳，以毫秒为单位
      uint64 sync_timestamp_ms;       //透传图像时间戳
      uint64 recv_timestamp_ms;       //接收到上游消息时间戳，ms
      boolean repeated_topic_flag;                //"0为非重复帧；1为重复帧"
      uint64 datacall_timestamp_ms;   //模块读取消息队列的时间
      uint64 frame_id;                //消息frame id
      boolean is_slice;              //Is this message sliced? Default is false
      uint32 slice_total_size;         //The total parts of the sliced message
      uint32 slice_index;              //The index of this message in sliced messge
    };
  };
};

#endif // KRIDER_IDL_INTERFACE_BASESTRUCTURE_BASESTRUCTURE
idl files like this   
other files are not necessary      
${TARGET_NAME} in cmake just like ${PROJECT_NAME}  in Fast-DDS-python/fastdds_python_examples/HelloWorldExample/CMakeLists.txt
 which is library for C++ types
